text/0000-build-config.md

```diff
@@ -0,0 +1,108 @@
+# Meta
+[meta]: #meta
+- Name: Build config
+- Start Date: 2022-08-29
+- Author(s): [samj1912](https://github.com/samj1912)
+- Status: Draft <!-- Acceptable values: Draft, Approved, On Hold, Superseded -->
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: (put "N/A" unless this replaces an existing RFC, then link to that RFC)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes an easy way to configure build images to allow specifying a `/cnb/config/env.build` CNB environment directory that allows updating the Buildpack `detect` and `build` environment based on the directory.
+
+
+# Definitions
+[definitions]: #definitions
+
+- CNB environment directory: A directory that follows the conventions as defined [here](https://github.com/buildpacks/spec/blob/main/buildpack.md#provided-by-the-buildpacks)
+- Operator: Owner of builders and stacks. See [CNB Operator guide](https://buildpacks.io/docs/operator-guide/).
+
+
+# Motivation
+[motivation]: #motivation
+
+Often times, especially in enterprise settings, organizations often have to update the buildpacks to use internal mirrors, proxies and other settings which can be configured easily with environment variables.
+
+Some examples include -
+- `GOPROXY`
+- `PIP_INDEX_URL`
+- `npm_config_registry`
+- `BUNDLE_MIRROR__URL`
+
+The buildpack logic in the Buildpacks largely remains the same, except these environment variables might need to be injected during the `build` and `detect` phases.
+
+The environment variables may ideally also take precendence over any user provided values to ensure that the operators have full control over their builders.
+
+# What it is
+[what-it-is]: #what-it-is
+
+The RFC proposes the introduction of the following directory `/cnb/config/env.build` in build images. The directory follows the same convention as a `CNB environment directory`. The notable difference is that the environment variables sourced from this directory are applied **AFTER** processing the user-provided platform environment variables i.e. they should have the highest precedence. These variables should be available during both `detect` and `build` phases (and the `generate` phase in the future).
+
+
+The operator can define this directory in the build image under `/cnb/config` or `CNB_CONFIG_DIR` if defined.
+
+# How it Works
+[how-it-works]: #how-it-works
+
+Reference implementation available at https://github.com/buildpacks/lifecycle/pull/899/files
+
+Examples - 
+
+Platform Enviroment varaible: `FOO=value`
+Build config: `FOO=another-value`
+Final value: `FOO=value`
```

Comment:
Is it possible to include a buildpack-provided value in this example? If I'm not mistaken, the order of application is:
buildpack -> platform (behavior: override) -> builder (behavior: configurable, with default `default`)

I recall that changing the order of application would be a larger breaking change - but what about changing the platform behavior so that it is:
buildpack -> platform (behavior: configurable, with default `default`) -> builder (behavior: configurable, with default `default`)

Platforms such as `pack` could add `.override` to env var files, to keep the same behavior.